[RFR] Pagination test case automation for stakeholders

nitishSr opened the pull request:
Test case automation script for pagination

ganeshhubale commented on cypress/integration/tests/controls/stakeholders/pagination.test.ts:
```diff
@@ -0,0 +1,169 @@
+        // Verify that only 10 items are displayed
+        cy.get("td[data-label=Email]").then(($rows) => {
+            var rowCount = $rows.length;
+            cy.wrap($rows.length).should("eq", 10);
```
Did you miss using that var rowCount or it has any other reason?
  `cy.wrap(rowCount).should("eq", 10);`

nitishSr replied:
Right, I missed that var. Good catch !

ganeshhubale commented on cypress/integration/tests/controls/stakeholders/pagination.test.ts:
```diff
@@ -0,0 +1,169 @@
+        // Verify that items less than or equal to 20 and greater than 10 are displayed
+        cy.get("td[data-label=Email]").then(($rows) => {
+            var rowCount = $rows.length;
+            cy.wrap($rows.length).should("be.lte", 20).and("be.gt", 10);
```
same here

ganeshhubale reviewed (changes requested):
@nitishSr Looks good. Minor change and request to cover one more scenario.

Steps:
1. Create 11 items and keep items per page value to 10
2. Go to 2nd page
3. Delete 11th item
4. **Expected**: It should show last page(in this case 1st page) elements. 

Note: it was bug which is now fixed - https://issues.redhat.com/browse/TACKLE-155

nitishSr commented:
> @nitishSr Looks good. Minor change and request to cover one more scenario.
> 
> Steps:
> 
> 1. Create 11 items and keep items per page value to 10
> 2. Go to 2nd page
> 3. Delete 11th item
> 4. **Expected**: It should show last page(in this case 1st page) elements.
> 
> Note: it was bug which is now fixed - https://issues.redhat.com/browse/TACKLE-155

Hi, do you want me to add a new test case or within any of current test case? Also, if new, what should be the name of that test case?

ganeshhubale commented:
> > @nitishSr Looks good. Minor change and request to cover one more scenario.
> > Steps:
> > 
> > 1. Create 11 items and keep items per page value to 10
> > 2. Go to 2nd page
> > 3. Delete 11th item
> > 4. **Expected**: It should show last page(in this case 1st page) elements.
> > 
> > Note: it was bug which is now fixed - https://issues.redhat.com/browse/TACKLE-155
> 
> Hi, do you want me to add a new test case or within any of current test case? Also, if new, what should be the name of that test case?

Yeah. It should be **new test case** as it covers bug. It will be helpful for us in case of regression testing.
Test case name - **Page load validation after deleting last element on page**

nitishSr commented:
> > > @nitishSr Looks good. Minor change and request to cover one more scenario.
> > > Steps:
> > > 
> > > 1. Create 11 items and keep items per page value to 10
> > > 2. Go to 2nd page
> > > 3. Delete 11th item
> > > 4. **Expected**: It should show last page(in this case 1st page) elements.
> > > 
> > > Note: it was bug which is now fixed - https://issues.redhat.com/browse/TACKLE-155
> > 
> > 
> > Hi, do you want me to add a new test case or within any of current test case? Also, if new, what should be the name of that test case?
> 
> Yeah. It should be **new test case** as it covers bug. It will be helpful for us in case of regression testing.
> Test case name - **Page load validation after deleting last element on page**

Added the test case.

ganeshhubale reviewed (approved):
Nice PR @nitishSr LGTM :+1:

sshveta commented on cypress/integration/tests/controls/stakeholders/pagination.test.ts:
```diff
@@ -0,0 +1,213 @@
+                    }
+                }
+            });
+    });
```
This can be done as :

     rowsTotest = 11;
     cy.get(appTable)
            .next()
            .then(($div) => {
                if (!$div.hasClass("pf-c-empty-state")) {
                    cy.get("td[data-label=Email]").then(($rows) => {
                        var rowCount = $rows.length;
                    }
            if (rowCount > rowstotest):
                  do nothing ;
           else : 
                  rowsToCreate = rowsTotest - rowCount
            for (let i = 0; i < rowsToCreate; i++) {
                        const stakeholder = new Stakeholders(data.getEmail(), data.getFullName());
                        stakeholder.create();
                        stakeholdersList.push(stakeholder);
                    }

nitishSr replied:
I tried it this way to use the loop only once. But value of var rowCount is not accessible outside if-else, since the scope of Promise gets over there itself.

sshveta reviewed (changes requested):
Looks Good Nitish , one minor comment

sshveta commented:
Please rebase.
